### PR TITLE
Fixes #32750 - make searchbar css looks like PF4

### PR DESCRIPTION
**Pull request by Ron-Lavi**
As the first step of moving it to PF4,
we can already override it with CSS and apply some changes to pages,

React pages are the only ones that would be affected at the moment,
legacy pages will still have the regular bootstrap3 design.

### CODE REVIEW

File: app/helpers/search_bar_helper.rb
Comment (Ron-Lavi) [inline app/helpers/search_bar_helper.rb]: should the legacy pages also use the new look or have the PF3 bootstrap design?

File: webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/AuditsPage.js
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/AuditsPage.js]: No need for icons in buttons, right? @marisvirik

File: webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/AuditsPage.js
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/AuditsPage.js]: Screenshot

File: webpack/assets/javascripts/react_app/routes/Models/ModelsPage/ModelsPage.js
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/routes/Models/ModelsPage/ModelsPage.js]: Screenshot (1)

cc @xprazak2

File: webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/AuditsPage.js
Comment (MariSvirik) [reply webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/AuditsPage.js]: which buttons are you referring to, @laviro ? But not really.

File: webpack/assets/javascripts/react_app/routes/Models/ModelsPage/ModelsPage.js
Comment (MariSvirik) [reply webpack/assets/javascripts/react_app/routes/Models/ModelsPage/ModelsPage.js]: use sentence case in a table header: 'Vendor class', 'Hardware model'
In buttons as well: 'Create model'

**Comment by MariaAga** [comment]
@laviro Can you add screenshots that show pf3 vs 4?

File: app/helpers/search_bar_helper.rb
Comment (MariaAga) [reply app/helpers/search_bar_helper.rb]: Can you add screenshots on how it looks? :)

Comment (Ron-Lavi) [reply app/helpers/search_bar_helper.rb]: If it will be the default this is how it would look:

gnome-shell-screenshot-928P80

maybe it will give us a push with replacing buttons all over :)

File: webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/AuditsPage.js
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/AuditsPage.js]: Before:
gnome-shell-screenshot-OHVG80

After:
gnome-shell-screenshot-XBAJ80

File: webpack/assets/javascripts/react_app/routes/Models/ModelsPage/ModelsPage.js
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/routes/Models/ModelsPage/ModelsPage.js]: Thanks, fixed in the second commit: https://github.com/theforeman/foreman/pull/8743/commits/b45316e9564dea889eb01eff841929d256a004ea

**Comment by Ron-Lavi** [comment]
> @laviro Can you add screenshots that show pf3 vs 4?

Audits page:
- before:
gnome-shell-screenshot-A8NW80

- after:


Models page:
- before:
gnome-shell-screenshot-O0RZ80

- after:

**Comment by MariaAga** [comment]
Waiting for a decision to be made about if we want to have a button to use the search or only keyboard 'enter'

**Comment by Ron-Lavi** [comment]
Any update design-wise on this prototype? @MariSvirik thanks 🙏🏼

**Comment by MariSvirik** [comment]
@laviro Not really.
PF4 is working on their own design ideations/research - cutting this complex search input monster into smaller chunks such as bookmarks/ supporting value-attribute pairs/ etc. But these parts aren't stable designs neither they are compatible with each other yet. 
For searching action in a search input, PF4 currently uses [an arrow icon](https://www.patternfly.org/v4/components/search-input/design-guidelines/#advanced-search) - which may not be super intuitive when presented with other icons as well.

Here are a couple of options with a bookmark button and a search button as well.
Screenshot 2021-09-20 at 11 08 34

Option 1/2 is more consistent with current PF4 search input designs. Option 4 is kind of what we currently use in some places in Satellite.
I would rather not use a search button at all - it's cleaner, less confusing.

**Comment by MariaAga** [comment]
@MariSvirik I think it's more confusing not having a search button, as without it the search is just a random text box and it won't search until the user will click enter.
Also option 1,2 button suggest that the page will go somewhere instead of search something which is not the case in foreman.

**Comment by Ron-Lavi** [comment]
I think we can add the button for now and just upgrade the look to PF4 so it won't block other parts of the page to be upgraded, e.g: the design of the current searchbar doesn't look good with new  PF4 buttons aside.

File: app/helpers/search_bar_helper.rb
Comment (tbrisker) [reply app/helpers/search_bar_helper.rb]: Maybe I missed something, why do we want the legacy pages to have a different style? IMHO one of the biggest strengths of the ruby helpers is that we can change everywhere at once and plugins don't need to care.

File: webpack/assets/javascripts/react_app/routes/Models/ModelsPage/ModelsPage.js
Comment (tbrisker) [reply webpack/assets/javascripts/react_app/routes/Models/ModelsPage/ModelsPage.js]: That doesn't seem related to this PR, please extract to a separate one (which would be a trivial merge).

File: webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/AuditsPage.js
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/AuditsPage.js]: This change doesn't seem to be related to the searchbar styling. I get that you want to make the buttons look similar, but that can be done in a different PR.

**Review by tbrisker — CHANGES_REQUESTED** [review]

**Comment by tbrisker** [comment]
Regarding the search button - I agree, option 4 looks better. I don't think removing it makes sense, users are used to having a button they can click. It makes sense to not have a button when a search field is used for a fast (client-side) filter that applies as they type, but when they need to press enter to search it makes sense to have the button at least for now. 

Also, test failures are related.

File: app/helpers/search_bar_helper.rb
Comment (Ron-Lavi) [reply app/helpers/search_bar_helper.rb]: because on old .erb pages the table and buttons on the page are still in bootstrap3 style,
I think it will just not look well in most of the pages until we switch those components to PF4 style,
when we'll do that, we can just tweak the `isPF4` prop.

File: webpack/assets/javascripts/react_app/routes/Models/ModelsPage/ModelsPage.js
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/routes/Models/ModelsPage/ModelsPage.js]: sure, no problems.

**Comment by Ron-Lavi** [comment]
This is how the search-bar looks now with PF4 styles:
pf4_search

and the current one for older pages:
Screenshot (29)

File: app/helpers/search_bar_helper.rb
Comment (tbrisker) [reply app/helpers/search_bar_helper.rb]: @MariSvirik what do you think? should we have different looking search fields on different pages, or consistent search fields everywhere, but with styling that is different from the rest of the page on legacy pages?
Personally, I tend to prefer the second, and gradually transform legacy pages, but not sure what is correct from UX standpoint. That approach also simplifies the code quite a bit.

**Comment by MariSvirik** [comment]
Can we just get rid of everything that's inside the search field?
/, Filter..., x and just write Search there?

It would be an easy fix to change 'Filter...' to 'Search'
Plus X button doesn't make sense if the search bar is empty. It can't do anything. It should appear if there is an input...

**Comment by MariaAga** [comment]
@laviro to fit the new design should this pr change the icon of the bookmarks?
Also, the new bookmark component is not complete as there were features added to the old one after the new was merged:
https://projects.theforeman.org/issues/32656

**Comment by Ron-Lavi** [comment]
> Can we just get rid of everything that's inside the search field? /, Filter..., x and just write Search there?
> 
> It would be an easy fix to change 'Filter...' to 'Search' Plus X button doesn't make sense if the search bar is empty. It can't do anything. It should appear if there is an input...

@MariSvirik I can try to display the `X` button only when there's some query,
and we can also rename the input's placeholder,
but as for removing the slash shortcut... it might need more feedback and research in the community - and I prefer we will do it properly when writing the component from scratch.

> @laviro to fit the new design should this pr change the icon of the bookmarks?
> Also, the new bookmark component is not complete as there were features added to the old one after the new was merged:
> https://projects.theforeman.org/issues/32656

@MariaAga if the PF4 bookmark isn't fully ready yet, I tend to lean towards just replacing the styles for the old PF3 bookmarks component until the new one is ready.

@tbrisker, let's make the searchbar consistent, I think it will look a bit weird at the start on old pages - but it might give us a good push towards PF4 and full react pages.

**Comment by MariaAga** [comment]
I agree about just changing the css for. Any way, opened https://github.com/theforeman/foreman/pull/8824 to fix icon+ add mange bookmarks.

File: app/helpers/search_bar_helper.rb
Comment (MariSvirik) [reply app/helpers/search_bar_helper.rb]: @tbrisker Based on the discussion with other designers we decided to go with this option:  'having the new component with pf4 styling on all pages, including legacy ones - this mean the search buttons will not match the action buttons until the buttons are updated to pf4 as well on those pages, but the search component will be the same everywhere' 

This would put more pressure on updating the legacy pages later on.

**Comment by MariSvirik** [comment]
@laviro sounds great. I can live with slash :D 
After discussion with our UX group, they opted for the approach to have a new search also on the legacy pages, which would pressure you to update the rest of the pages pretty soon.

**Comment by Ron-Lavi** [comment]
> Any way, opened #8824 to fix icon+ add mange bookmarks.

NIce @MariaAga, that was fast!
actually, I tried manipulating the old bookmark to look like PF4 one, but it took too much work and too many file changes
so I used the commit from #8824 and it works great!

this is how the search bar looks at the moment:
Screenshot (32)

Also, I wanted to remove the old bookmarks folder, though it seems the new PF4 bookmarks is trying to import components from there, so let's do it as a follow up and keep this PR small.

any other thoughts or concerns?

**Comment by tbrisker** [comment]
@laviro please rebase, the other PR has been merged.

**Comment by Ron-Lavi** [comment]
Rebased, thanks @MariaAga @tbrisker !
@MariSvirik I also removed the dots and the italic style to the placeholder:
Screenshot (33)

**Comment by MariaAga** [comment]
@laviro  Great to see that the bookmarks work with the "new" search!
The new search looks short, and it's not taking it's full length of 500px even, can it be longer?
Screenshot from 2021-10-11 14-07-26
Screenshot from 2021-10-11 14-06-02

**Comment by Ron-Lavi** [comment]
Thanks @MariaAga ! makes sense, I will try to make it larger

**Comment by tbrisker** [comment]
@laviro test failures are related

**Comment by Ron-Lavi** [comment]
@MariaAga I just fixed the search bar width:

gnome-shell-screenshot-99Z0A1

File: webpack/assets/javascripts/react_app/components/PF4/Bookmarks/Bookmarks.js
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/PF4/Bookmarks/Bookmarks.js:15]: Did the bookmarks title got deleted? I think it was a good idea to specify that " this icon is a button to bookmarks"

Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/PF4/Bookmarks/Bookmarks.js:15]: probably during the rebase

Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/PF4/Bookmarks/Bookmarks.js:15]: added it again 👍🏼

**Comment by Ron-Lavi** [comment]
ready for another review round :)

**Review by MariaAga — APPROVED** [review]
Looka good to me :)
@tbrisker any other concerns?

**Review by tbrisker — APPROVED** [review]
Thanks @laviro and @MariaAga !

**Pull request merged**
